### ISSUE TITLE
tooltip is not hidding 

### ISSUE THREAD

**Issue by wirelessmundi** [novus/nvd3#1705]
with master or 1.8.3 version tooltip os not hidding when the mouse leaves the graph area, so multiple tooltips are shown in the graph area.

![tooltip-issue](https://cloud.githubusercontent.com/assets/13020619/16418675/27b6e102-3d4b-11e6-96ff-093b0187aca7.png)

before i had version 1.8.1 and its fine. 

The code is quite basic:

```
d3.json(jsonurldata, function(error, data) {
    if (error) return console.warn(error);

    if (!data.timeformat) timeformat = '%x';
    else timeformat = data.timeformat;

    nv.addGraph(function() {
        var chart = nv.models.stackedAreaChart()
            .margin({'right': 50, 'bottom': 30})
            .xScale(d3.time.scale())
            .useInteractiveGuideline(true)
            .showControls(false)
            .clipEdge(true);

        if (color) chart.color(color);

        // x time scale
        chart.xAxis.scale(d3.time.scale());
        chart.xAxis.tickFormat(function(d) { 
            return d3.time.format(timeformat)(new Date(d)) 
        });
        chart.interactiveLayer.tooltip.headerFormatter(function (d) { return d; });
        chart.xAxis.ticks(5);
        chart.yAxis.tickFormat(d3.format(',.2f'));

        d3.select(container)
            .datum(data.series)
            .transition().duration(1200)
            .call(chart);

        nv.utils.windowResize(chart.update);
        return chart;
    });
});
```

Is there any new parameter that i should use instead? 

EDIT: this issue is only present in firefox (tested with version 47.0) using linux. Not really sure what could be different.. 

Thanks.

**Comment by wirelessmundi** [comment]
I tried the solution #1463, which solves the problem on scrool down, but for the mouse out event i have the same result, the tooltip get stuck in the graph.. 

d3.select(window).on("mouseout."+chart.id(), function() {
	console.log("Mouse OUT");
	chart.tooltip.hidden(true);
	chart.interactiveLayer.tooltip.hidden(true);
});

**Comment by liquidpele** [comment]
Latest version has some tooltip fixes, see if your issue still exists... 

If it does, please make a jsfiddle demonstrating it (since I don't see the issue) using the latest nvd3 in master (you can link to latest using rawgit.com)

**Comment by wirelessmundi** [comment]
it work as expected with master. 

Using version 1.8.4 i still have the same issue, but only when using firefox 47 with debian 8.

jsfiddle here: https://jsfiddle.net/69mtzt1w/8/

![screenshot from 2016-07-05 10 17 59](https://cloud.githubusercontent.com/assets/13020619/16578282/f39840b0-4299-11e6-8bae-394d321d0ed9.png)

**Comment by liquidpele** [comment]
Wait, it works in master but not in 1.8.4??  That's odd, I thought I merged in all the tooltip stuff before I cut the 1.8.4. release...

**Comment by wirelessmundi** [comment]
yeahp.. at the first time when the page load its ok, is only after hitting F5 multiple times that triggers the issue.

**Comment by cmbreki** [comment]
Hello, i'm facing the exact same problem. I tried with the v.1.8.4  or the master version but the tooltip doesn't work properly on firefox.
Has there been any progress on the issue ?

**Comment by mehranshakeri** [comment]
same problem. I wonder if it's nvd3 bug or FireFox issue in rendering the page. When you try to find DOM element, there is no element for those visible tooltips and they are not even a full tooltip. It looks like a rendering issue of browser!

**Comment by mehranshakeri** [comment]
Well, I guess in my case it was my graphic driver or OS problem. I have ubuntu 16.04 with Radeon VGA. Using other PCs with windows this bug doesn't happen at all!!!

**Comment by mke1ms** [comment]
Same here, problem only occurs on Firefox (47.0) on Ubuntu 14.04, Chrome works fine

**Comment by liquidpele** [comment]
Well that's just weird.  Are there any javascript errors or anything?

**Comment by joserocha3** [comment]
@liquidpele I don't get any

**Comment by mke1ms** [comment]
@liquidpele Nope, no errors, just bits and pieces of tooltips all over the screen
![tooltip](https://cloud.githubusercontent.com/assets/7792223/17814732/7f577390-6631-11e6-9a50-32071c0efa9d.png)

**Comment by mehranshakeri** [comment]
No JS error. It seems it's they way FireFox uses graphic card.

**Comment by liquidpele** [comment]
Wow... so not even the full div, just part of it shows... I'm not sure that's something we can fix... it looks like a firefox rendering bug...  if anyone can figure out a workaround we can implement let me know though.   Maybe even submit it as a bug report to bugzilla?

**Comment by slestang** [comment]
It is a Firefox bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1280060
It should be fix with Firefox 49.
I have a branch to work around the bug but it isn't very clean: https://github.com/slestang/nvd3/tree/workaround_firefox_bug_tooltip

**Comment by liquidpele** [comment]
According to their release calendar, 49 will be released Sep 13th.  

https://wiki.mozilla.org/RapidRelease/Calendar

That's soon enough to just wait I think.

**Comment by rodrigorrm** [comment]
I use Chrome 55.0.2883.87 (64-bit) on Ubuntu 16.04 and have the same issue. Is there any development on this issue?

**Comment by pgrm** [comment]
I see a similar issue again on Firefox 54, IE and chrome work for me.

**Comment by hajdbo** [comment]
related to #1886

**Comment by stla** [comment]
I have a similar bug when including a NVD3 multibarchart in **Power BI**:

![nvd3_bug_tooltips](https://user-images.githubusercontent.com/4466543/128631856-7241285b-3dc1-41af-8ec2-ddcd0d9c8325.png)

Does someone has a solution now?
